bot/decorators.py

```diff
@@ -105,15 +98,15 @@ def wrap(func):
     return wrap
 
 
-def redirect_output(destination_channel: int, bypass_roles: typing.Container[int] = None):
-    """
-    Changes the channel in the context of the command to redirect the output
-    to a certain channel, unless the author has a role to bypass redirection
+def redirect_output(destination_channel: int, bypass_roles: Container[int] = None) -> Callable:
     """
+    Changes the channel in the context of the command to redirect the output to a certain channel.
 
-    def wrap(func):
+    Redirect is bypassed if the author has a role to bypass redirection.
+    """
+    def wrap(func: Callable) -> Callable:
         @wraps(func)
-        async def inner(self, ctx, *args, **kwargs):
+        async def inner(self: Callable, ctx: Context, *args, **kwargs) -> None:
```

Comment:
I made a mistake during my previous review, this function does have two paths that include `return await func`, so I think we should make this `Optional[Any]` or just `Any`, since `Any` is compatible with `None`. 
```suggestion
        async def inner(self: Callable, ctx: Context, *args, **kwargs) -> Any:
```